rust/rope/Cargo.toml

```diff
@@ -10,11 +10,15 @@ edition = '2018'
 [dependencies]
 bytecount = "0.5"
 memchr = "2.0"
-serde = "1.0"
-serde_derive = "1.0"
+serde = { version="1.0", optional=true }
+serde_derive = { version="1.0", optional=true }
 unicode-segmentation = "1.2.1"
 regex = "1.0"
 
 [dev-dependencies]
 serde_test = "^1.0"
 serde_json = "1.0"
+
+[features]
+default = ["dese"]
```

Comment:
We can't call the feature `serde`, Cargo complains that we can have a feature whose name is the same as one of our deps.

Reply:
right. Bikeshed, but maybe we call our feature `serde_impls`? Or otherwise something else where the name clearly describes the intent.

Reply:
I would argue against this feature being default, and require its users to opt in explicitly. Otherwise you have to do no-default-features, which is not pleasant.

Reply:
Hm, I personally prefer featues being enabled by default, but sure, I can make it disabled by default.

Reply:
Yea I can see both ways. The case against is that the serde stuff hurts compile times and code size and is only needed in use cases like `xi`, so we should give you the faster, smaller version by default. The case *for* is that we want to give you everything up front, and if you realize you don't need serde you can disable. I'm happy to defer to raph.

Reply:
Also, I think this can be named "serde" by doing this:

    serde = { version="1.0", optional=true, features=["derive"] }
    serde_derive = { version="1.0", optional=true }

And not having a separate features section at all

Reply:
Oh, that does work, thanks!